components/sbm-recipes-boot-upgrade/src/test/java/org/springframework/sbm/boot/upgrade_27_30/ExplicitVersionTo30Test.java

```diff
@@ -0,0 +1,127 @@
+                        <dependency>
+                            <groupId>io.dropwizard.metrics</groupId>
+                            <artifactId>metrics-annotation</artifactId>
+                            <version>4.2.8</version>
+                        </dependency>
+                        <dependency>
+                            <groupId>org.springframework.boot</groupId>
+                            <artifactId>spring-boot-starter-test</artifactId>
+                            <version>2.7.3</version>
+                            <scope>test</scope>
+                        </dependency>
+                    </dependencies>
+                </project>
+                """);
+
+        List<Result> result = recipe.run(documentList, ctx);
+
+        assertThat(result).hasSize(1);
+
+        assertThat(result.get(0).getAfter().printAll())
+                .isEqualTo("""
+                    <?xml version="1.0" encoding="UTF-8"?>
+                    <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                             xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+                        <modelVersion>4.0.0</modelVersion>
+                        <groupId>com.example</groupId>
+                        <artifactId>explicit-deps-app</artifactId>
+                        <version>0.0.1-SNAPSHOT</version>
+                        <name>explicit-deps-app</name>
+                        <description>explicit-deps-app</description>
+                        <properties>
+                            <java.version>17</java.version>
+                            <maven.compiler.source>17</maven.compiler.source>
+                            <maven.compiler.target>17</maven.compiler.target>
+                        </properties>
+                    
+                        <dependencies>
+                            <dependency>
+                                <groupId>org.springframework.boot</groupId>
+                                <artifactId>spring-boot-starter-web</artifactId>
+                                <version>3.0.0-M3</version>
+                            </dependency>
+                            <dependency>
+                                <groupId>io.dropwizard.metrics</groupId>
+                                <artifactId>metrics-annotation</artifactId>
+                                <version>4.2.9</version>
+                            </dependency>
+                            <dependency>
+                                <groupId>org.springframework.boot</groupId>
+                                <artifactId>spring-boot-starter-test</artifactId>
+                                <version>3.0.0-M3</version>
+                                <scope>test</scope>
+                            </dependency>
+                        </dependencies>
+                    </project>
+                    """);
```

Comment:
Could you add an assertion accessing the model here?
E.g. `result.get(0).getAfter().getMarkers().findFirst(MavenResolutionResult.class).getDependencies().get(0).getVersion) isEqualTo "3.0.0-M3"` ? Are the changes reflected in the model?

Reply:
Yes, updated to assertion and added to UpgradeUnmanagedSpringProjectTest to verify model as well

Reply:
Thanks